public/app/plugins/datasource/tempo/resultTransformer.ts

```diff
@@ -220,9 +220,11 @@ function getSpanTags(
     span.kind !== undefined &&
     span.kind !== collectorTypes.opentelemetryProto.trace.v1.Span.SpanKind.SPAN_KIND_INTERNAL
   ) {
+    // collectorTypes...SpanKind includes "SPAN_KIND_UNSPECIFIED" at 0 so we need to subtract 1 from
+    // the index to map to the correct value in the SpanKind enum
     spanTags.push({
       key: 'span.kind',
-      value: SpanKind[collectorTypes.opentelemetryProto.trace.v1.Span.SpanKind[span.kind] as any].toLowerCase(),
+      value: SpanKind[(collectorTypes.opentelemetryProto.trace.v1.Span.SpanKind[span.kind] as any) - 1].toLowerCase(),
```

Comment:
Bit unsure why this code looks like this. It seems to me we could just do
```
spanTags.push({
      key: 'span.kind',
      value: span.kind.toString().toLowerCase().split('_').last()
})
```
or something like that as it seems like we just need to convert the enum string values so they are a bit more readable. Also not sure why SPAN_KIND_INTERNAL isn't added? If we are processing the span why not add its kind here? @zoltanbedi do you maybe remember why we do it like this?

Reply:
I think we lose a bit of safety by just converting the string although since it's a tag, it's probably not a big deal to have any string as the value. That would simplify the code a bit.

INTERNAL is the default span kind so maybe that's why we don't add the tag? Seems like we should still just add it.

Reply:
>I think we lose a bit of safety by just converting the string 

What do you mean? That the enum value name could change? I mean yeah probably should fallback to not split() or something just in case to still get something meaningful but don't see many other issues. If the definitions change the current approach could probably fail even more because we are not really doing type-checked access to those enums anyway.

We can still filter out the `internal` tag if needed, not sure if it is important or not.

Reply:
I was thinking that by using the enum, we're restricting it to a set of valid values. Without the enum, we're allowing any value which I guess is fine. True that if the definitions change, it'd break. 

I'll switch to just going off of the string value and add some more error handling. I'll also remove the INTERNAL condition unless Zoltan confirms we need it.

Reply:
> I was thinking that by using the enum, we're restricting it to a set of valid values.

Just to clarify, aren't both `SpanKind` and `collectorTypes.opentelemetryProto.trace.v1.Span.SpanKind` coming from OpenTelemetry types? Do they have different meanings? It seems to me like a definitions bug that they have different values tbh.

Reply:
Yeah, they do differ because one is from the js-api and the other from the collector. It does seem like a bug that they differ - that difference is what caused this bug in the first place

Reply:
> Also not sure why SPAN_KIND_INTERNAL isn't added? If we are processing the span why not add its kind here? @zoltanbedi do you maybe remember why we do it like this?

I think I just missed it.

Reply:
thanks, in that case, let's merge this PR, I don't think having that additional tag there would make anything worse and seem more consistent this way.